### ISSUE TITLE
camelcase rule doesn't ignore default parameter assignment when disabled

### ISSUE THREAD

**Issue by cdanielsen** [eslint/eslint#8511]
**Tell us about your environment**

**ESLint Version:** 3.19.0
**Node Version:** 6.9.4
**npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?** Default

**Please show your full configuration:**

```
module.exports = {
    "extends": "standard",
    "plugins": [
        "standard",
        "promise"
    ],
    "parserOptions": {
      "ecmaVersion": 6
    },
    "env": {
      "node": true
    },
    "rules": {
      "semi": ["error", "always"],
      "camelcase": [2, {"properties": "never"}]
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

Disabling the `camelcase` rule does not appear to respect ES6 default parameter assignment when used with object destructuring syntax.

Here's a simple example:

```
// example.js

function snakeCase ({
  param_1 = 'default value',
  param_2 = 'another default'
} = {}) {
  // do stuff
}

module.exports = snakeCase;

```

**What did you expect to happen?**

Eslint should not care that `param_1` and `param_2` are using snake case instead of camelCase as the rule is disabled

**What actually happened? Please include the actual, raw output from ESLint.**

```
error  Identifier 'param_1' is not in camel case                            camelcase
error  Identifier 'param_2' is not in camel case                            camelcase
```

**Comment by cdanielsen** [comment]
Not sure if this is a bug or a rule change request; thanks for reviewing

**Comment by vitorbal** [comment]
Thanks for the report, @cdanielsen! I can reproduce this. Note that the following works:

```js
function snakeCase ({
  param_1,
  param_2
} = {}) {
  // do stuff
}
```
I think the default assignment there is what's throwing the rule off.

**Comment by vitorbal** [comment]
I'll work on this.
